### PR TITLE
Performance improvements for Unitful conversion

### PR DESCRIPTION
**Pull request by Deduction42**
Fix for issue #139, major (~500x) performance improvements when converting from Unitful to DynamicUnits, minor improvements to reverse case.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 /Manifest.toml
 /docs/Manifest.toml
 /docs/build/
+/.vscode*
```
Comment (MilesCranmer) [inline .gitignore:7]:
```suggestion
```

**Comment by Deduction42** [comment]
Is it passing the tests now? I see a green checkmark on "Benchmark a pull request" workflow. I has one workflow awaiting approval. Is there something else I need to look at for the tests?

**Comment by Deduction42** [comment]
Okay, I think I know what happened. I made some changes to Main on my side and  I didn't know that the pull request tracked the current version of my repo. I revered those changes so that only the changes we discussed were made. Sorry about that.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (MilesCranmer) [inline .vscode/settings.json:2]: this file should be deleted

File: ext/DynamicQuantitiesUnitfulExt.jl [hunk 3]
```diff
@@ -53,29 +53,22 @@ for (_, _, Q) in ABSTRACT_QUANTITY_TYPES
...
+    D == :Mass          && return DimType(mass = dims.power)
+    D == :Time          && return DimType(time = dims.power)
+    D == :Current       && return DimType(current = dims.power)
+    D == :Temperature   && return DimType(temperature  = dims.power)
+    D == :Luminosity    && return DimType(luminosity  = dims.power)
+    D == :Amount        && return DimType(amount = dims.power)
+    error("Unknown dimension: $D")
 end
+
+end
```
Comment (MilesCranmer) [inline ext/DynamicQuantitiesUnitfulExt.jl:74]:
```suggestion
end

```

File: ext/DynamicQuantitiesUnitfulExt.jl [hunk 4]
```diff
@@ -53,29 +53,22 @@ for (_, _, Q) in ABSTRACT_QUANTITY_TYPES
...
-
+function _dynamic_dimension(::Type{R}, dims::Unitful.Dimension{D}) where {R,D}
+    DimType = DynamicQuantities.Dimensions{R}
+    D == :Length        && return DimType(length = dims.power)
+    D == :Mass          && return DimType(mass = dims.power)
+    D == :Time          && return DimType(time = dims.power)
+    D == :Current       && return DimType(current = dims.power)
+    D == :Temperature   && return DimType(temperature  = dims.power)
+    D == :Luminosity    && return DimType(luminosity  = dims.power)
+    D == :Amount        && return DimType(amount = dims.power)
```
Comment (MilesCranmer) [inline ext/DynamicQuantitiesUnitfulExt.jl:70]:
```suggestion
    D == :Length && return DimType(length=dims.power)
    D == :Mass && return DimType(mass=dims.power)
    D == :Time && return DimType(time=dims.power)
    D == :Current && return DimType(current=dims.power)
    D == :Temperature && return DimType(temperature=dims.power)
    D == :Luminosity && return DimType(luminosity=dims.power)
    D == :Amount && return DimType(amount=dims.power)
```

File: .vscode/settings.json (hunk 2 above)
Comment (MilesCranmer) [inline .vscode/settings.json:2]: This file is still here; not sure if you meant to remove it

(It won't let me edit your PR directly so can't help sadly)

**Comment by MilesCranmer** [comment]
Thanks for the PR!

**Comment by Deduction42** [comment]
No problem!

**Pull request merged**